File: analise/paths.py
"""Caminhos padrão do projeto.

O diretório de dados pode ser redirecionado por PORTFOLIO_DATA_DIR — é assim
que os testes trabalham num diretório temporário, e é a mesma variável que a
aplicação Node respeita.
"""

import os
from pathlib import Path

BASE_DIR = Path(__file__).resolve().parent.parent


def data_dir() -> Path:
    return Path(os.getenv("PORTFOLIO_DATA_DIR") or (BASE_DIR / "data"))


def history_dir() -> Path:
    return data_dir() / "history"


def portfolio_file() -> Path:
    return data_dir() / "portfolio.json"


def cache_file() -> Path:
    return data_dir() / "cache.json"


def last_analysis_file() -> Path:
    return data_dir() / "last_analysis.json"


def telegram_enviados_file() -> Path:
    """Estado de deduplicação do Telegram — ver `analise.estado_envio`."""
    return data_dir() / "telegram_enviados.json"


def execucoes_file() -> Path:
    """Log de execuções. Ao contrário do arquivo do dia, nunca é sobrescrito:
    cada rodada acrescenta uma linha, e por isso uma falha continua visível
    depois de uma execução seguinte bem-sucedida."""
    return data_dir() / "execucoes.json"


def garantir_diretorios() -> None:
    """Cria os diretórios de dados. Chamado antes de qualquer escrita."""
    history_dir().mkdir(parents=True, exist_ok=True)

File: src/config/paths.js
/**
 * Caminhos padrão do projeto.
 *
 * Todos são resolvidos a cada chamada (e não no carregamento do módulo) para
 * que os testes possam apontar PORTFOLIO_DATA_DIR para um diretório temporário.
 */

import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

export const BASE_DIR = path.resolve(path.dirname(fileURLToPath(import.meta.url)), "..", "..");

export const WEB_DIR = path.join(BASE_DIR, "web");
export const SCRIPTS_DIR = path.join(BASE_DIR, "scripts");
export const SCRIPT_ANALISE = path.join(SCRIPTS_DIR, "analisar.py");

export function dataDir() {
  return process.env.PORTFOLIO_DATA_DIR || path.join(BASE_DIR, "data");
}

export function historyDir() {
  return path.join(dataDir(), "history");
}

export function portfolioFile() {
  return path.join(dataDir(), "portfolio.json");
}

export function cacheFile() {
  return path.join(dataDir(), "cache.json");
}

export function lastAnalysisFile() {
  return path.join(dataDir(), "last_analysis.json");
}

/** Log de execuções, escrito pelo Python a cada rodada (uma linha por execução). */
export function execucoesFile() {
  return path.join(dataDir(), "execucoes.json");
}

/** Usuários do login e segredo de sessão — só o Node lê e escreve. */
export function usuariosFile() {
  return path.join(dataDir(), "usuarios.json");
}

/** Cria os diretórios de dados. Chamado antes de qualquer escrita. */
export function garantirDiretorios() {
  fs.mkdirSync(historyDir(), { recursive: true });
}
